Build failure: kak-lsp

Part of #332957.

### Steps To Reproduce

Steps to reproduce the behavior:
1. build kak-lsp

### Build log

```
kakoune-lsp> error[E0282]: type annotations needed for `Box<_>`
kakoune-lsp>   --> /build/kakoune-lsp-17.1.1-vendor.tar.gz/time/src/format_description/parse/mod.rs:83:9
kakoune-lsp>    |
kakoune-lsp> 83 |     let items = format_items
kakoune-lsp>    |         ^^^^^
kakoune-lsp> ...
kakoune-lsp> 86 |     Ok(items.into())
kakoune-lsp>    |              ---- type must be known at this point
kakoune-lsp>    |
kakoune-lsp> help: consider giving `items` an explicit type, where the placeholders `_` are specified
kakoune-lsp>    |
kakoune-lsp> 83 |     let items: Box<_> = format_items
kakoune-lsp>    |              ++++++++

```

### Additional context

Fails on unstable and master, tested with ee014a8258713744d11a10f9be56fcd94f133580

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@spacekookie @poweredbypie

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.11 (Vicuna), 24.11.20240702.7d4534c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ka/kakoune-lsp/package.nix
+++ b/pkgs/by-name/ka/kakoune-lsp/package.nix
@@ -12,18 +12,18 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "kakoune-lsp";
-  version = "17.1.1";
+  version = "17.1.2";
 
   src = fetchFromGitHub {
     owner = "kakoune-lsp";
     repo = "kakoune-lsp";
     rev = "v${version}";
-    sha256 = "sha256-XBH2pMDiHJNXrx90Lt0IcsbMFUM+X7GAHgiHpdlIdR4=";
+    sha256 = "sha256-NZDp98Ne6z7DlJ2vZiqGzw5ukusOkEjb+eyvmxB+IKI=";
   };
 
   patches = [ (replaceVars ./Hardcode-perl.patch { inherit perl; }) ];
 
-  cargoHash = "sha256-Yi+T+9E3Wvce4kDLsRgZ07RAGLrq7dkinKpvvGeLeS0=";
+  cargoHash = "sha256-QonOqdcdp1vbxzLnF46X0DLVay2Up1LvHZ/ZZ04LqlE=";
 
   buildInputs = lib.optionals stdenv.isDarwin [
     CoreServices

